docs/source/blueprints/BP005.rst

```diff
@@ -0,0 +1,275 @@
+interaction with system level features and interfaces.  As such, if a user
+needs to interact with such a system level feature or interface, he/she may
+propose a new module.
+
+For instance, if a user finds himself/herself constantly interacting with disk
+partitions, he/she may want to propose a “partitions” module that provides
+basic APIs for that. In this case it makes sense to have it as part of this
+central repository.
+
+However, if the utility intends to ease the interaction with an esoteric
+feature, system or device, this may not be a good place for it.
+
+As a general rule, if a system level feature or interface is generically
+available to a multitude of users, such as the partition example above, it is a
+good candidate for this repository. Common sense is much appreciated here.
+
+Besides that, the proposed utility must have at least one MAINTAINER that will
+be responsible for keeping track of Pull Requests, tests and quality
+requirements as described below:
+
+Quality requirements
+--------------------
+
+It’s expected that both new modules and changes to existing modules will need
+to observe a full set of quality requirements and general standards that will
+either be described (documentation) and enforced as automated checks as much as
+possible.
+
+Among the topics related to the quality requirements and standards, the
+contribution documentation and automated checks will cover, at least:
+
+* Logical, stylistic, analytical and formatter checks
+
+  * Docstring checks
+  * Pylint
+  * PEP-8, PEP-257, etc…
+
+* Unit and Functional Tests Coverage
+
+  Due to the current code quality and given this is a new repository we would
+  like to reach at least 90% of coverage. This way, users writing tests relying
+  on these libraries will have a guarantee of a good level of correctness and
+  quality, redirecting the debugging and troubleshooting efforts away from the
+  well tested libraries.
+
+The new standalone repository will have predefined pipelines to run those
+checks, so new libraries will be automatically enforced to newly introduced
+libraries.
+
+Metadata description
+--------------------
+
+Every module will need to provide a description including:
+
+* Utility description
+* The utility category (ie: network, cpu, security, etc..)
+* Maintainers information
+  * Main contact (We need a contact point, a single maintainer or a group)
+  * Maintainer Group members (this will be used to block merges)
+* Supported Platforms
+* Supported execution method (Local vs Remote)
+
+For instance, the textual category and description will allow users to search
+for modules based on their interest.  Also, our pipelines will parse that data
+field containing the list of supported platforms, and will trigger tests on
+those platforms. Users will then be able to see the results of those tests on
+each of the platforms defined, giving them quality assurance and coverage
```

Comment:
Not directly related to the search comments here but I hope you notice how your variant definitions start looking more and more like Avocado VT-s, the only difference being that instead we use containers on Github :o)

Reply:
> And yes, we are now able to do the "distro" part very easily with avocado run --nrunner-spawner=podman --spawner-podman-image=$distro.

This is definitely nice but seems like we can control a single parameter in comparison to the Cartesian products of parameters and OS variations we can with Avocado VT. I actually had a vague idea to extract the Cartesian config and help our to provide it as Avocado varianter plugin if anyone is interested (not meant to simplify the learning curve unfortunately but it could provide the variant expression power many of us have grown accustomed to and can still be used with simpler configurations). If we were to migrate the main varianter for Avocado VT and most of its utilities, it will be left with just the framework of VM classes and VM isolation environments code which could be more easily digested and analyzed for potential replacement or simplification refactoring.

But I won't kidnap the topic here, my point is that we want *both* easy entry for adopters and expression power for continuous users.

> BTW, this reminds me that I owe some TLC in the LXC spawner PR (my apologies!).

No worries, I will convert it away from draft today so that it is more visible and make your lives simpler, I am aware that the draft status is not compatible with the workflow of the Avocado maintainers. I will also do my best to assist you with maintenance in the long term if you need it as this spawner will turn into core component in our Avocado testing.